marshmallow_oneofschema/one_of_schema.py

```diff
@@ -144,7 +144,13 @@ def _load(self, data, partial=None):
                 self.type_field: ['Missing data for required field.']
             })
 
-        type_schema = self.type_schemas.get(data_type)
+        try:
+            type_schema = self.type_schemas.get(data_type)
+        except TypeError as exc:
+            # data_type could be unhashable
+            return UnmarshalResult({}, {
+                self.type_field: [str(exc)]
```

Comment:
Can you change error message to something like "Invalid type value" instead of showing exception message